cocotb/build_libs.py

```diff
@@ -0,0 +1,420 @@
+logger = logging.getLogger(__name__)
+
+
+class build_ext(_build_ext):
+
+    # Needed for Windows to not assume python module (generate interface in def file)
+    def get_export_symbols(self, ext):
+        return None
+
+    # For proper cocotb library nameing, based on https://github.com/cython/cython/issues/1740
+    # TODO: for the simulator python library, we should be leaving this suffix in place
+    def get_ext_filename(self, ext_name):
+        """
+        Like the base class method, but removes the ``.cpython-36m-x86_64-linux-gnu.`` part before the extension.
+
+        Also replaces ``.pyd`` with ``.dll`` on windows.
+        """
+
+        filename = _build_ext.get_ext_filename(self, ext_name)
+
+        head, tail = os.path.split(filename)
+        tail_split = tail.split(".")
+
+        if os.name == "nt":
+            ext_name = "dll"
+        else:
+            ext_name = "so"
+
+        filename_short = os.path.join(head, tail_split[0] + "." + ext_name)
+
+        # icarus requires vpl extension
+        if filename.find("icarus") >= 0:
+            filename_short = filename_short.replace("libvpi.so", "gpivpi.vpl")
+            filename_short = filename_short.replace("libvpi.dll", "gpivpi.vpl")
+
+        return filename_short
+
+    # Add extra library_dirs path
+    def finalize_options(self):
+        _build_ext.finalize_options(self)
+
+        for ext in self.extensions:
+            ext.library_dirs.append(
+                os.path.join(self.build_lib, os.path.dirname(ext._full_name))
+            )
+
+    # copy libs into proper directory in develop
+    def copy_extensions_to_source(self):
+        build_py = self.get_finalized_command("build_py")
+        for ext in self.extensions:
+            fullname = self.get_ext_fullname(ext.name)
+            filename = self.get_ext_filename(fullname)
+            modpath = fullname.split(".")
+            package = ".".join(modpath[:-1])
+            package_dir = build_py.get_package_dir(package)
+            dest_filename = os.path.join(package_dir, filename)
+            src_filename = os.path.join(self.build_lib, filename)
+
+            os.makedirs(os.path.dirname(dest_filename), exist_ok=True)
+
+            copy_file(
+                src_filename, dest_filename, verbose=self.verbose, dry_run=self.dry_run
+            )
+            if ext._needs_stub:
+                self.write_stub(package_dir or os.curdir, ext, True)
+
+
+def _extra_link_args(lib_name):
+    if sys.platform == "darwin":
+        return ["-Wl,-install_name,@loader_path/%s.so" % lib_name]
+    else:
+        return []
+
+
+_base_warns = ["-Wall", "-Wextra", "-Wconversion", "-Wcast-qual", "-Wwrite-strings"]
+_cc_warns = _base_warns + ["-Wstrict-prototypes", "-Waggregate-return"]
+_ccx_warns = _base_warns + ["-Wnon-virtual-dtor", "-Woverloaded-virtual"]
+
+# TODO Enable warnings when fixed
+_extra_cc_compile_args = ["-std=gnu99"]  # + _cc_warns
+_extra_cxx_compile_args = ["-std=c++11"]  # + _ccx_warns
```

Comment:
Is this now fixed? If not, then can you comment out the specific warnings that are failing, so that CI can enforce we don't regress on the ones that are fixed.